backend/app/models/delivery.py

```
@@ -216,6 +216,24 @@ def __init__(self, **kwargs: object) -> None:
+
+
+class ProjectAutomationRun(LoopNode):
+    """One scheduled or manual occurrence of a project automation."""
+
+    __mapper_args__ = {"polymorphic_identity": "automation_run"}
+
+
+class ProjectAutomationBugLink(LoopNode):
+    """Stable bug identity used to make automation-created tasks idempotent."""
+
+    __mapper_args__ = {"polymorphic_identity": "automation_bug_link"}
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)